Feature unsized-locals changed with latest nightly

## Issue

The nightly feature required here: 
https://github.com/kompics/kompact/blob/b3d31ba48abd1a131c3c1815b8f1a74c07a5f804/core/src/utils.rs#L608-L669
And turned on optionally there:
```
core/src/lib.rs
74:#![cfg_attr(nightly, feature(unsized_locals))]
```

breaks with nightly 2020-10-29 because of https://github.com/rust-lang/rust/pull/78152

`feature(unsized_locals)` was split into two disjoint features:
- the `feature(unsized_locals)` which is marked as [incomplete](https://doc.rust-lang.org/beta/rustc/lints/listing/warn-by-default.html#incomplete-features)
- `feature(unsized_fn_params)` which isn't, and which is what Kompact is actually using here.

So post nightly 2020-10-29, Kompact is not activating the feature it actually needs.

## Steps to reproduce

```
cargo +nightly-2020-10-29-x86_64-unknown-linux-gnu check
```

possibly preceded by a `rustup install` to install the nightly in question.
Witness [the following output](https://gist.github.com/b2a5ff88d6176e7d379743623d96a83a):
```
error[E0277]: the size for values of type `Self` cannot be known at compilation time
   --> /home/huitseeker/.cargo/git/checkouts/kompact-8e3803f644d2dc0c/b3d31ba/core/src/utils.rs:621:32
    |
621 |         fn for_each_with<T, F>(mut self, t: T, mut f: F)
    |                                ^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: unsized fn params are gated as an unstable feature
help: consider further restricting `Self`
    |
624 |             F: FnMut(Self::Item, T), Self: Sized
    |                                    ^^^^^^^^^^^^^
help: function arguments must have a statically known size, borrowed types always have a known size
    |
621 |         fn for_each_with<T, F>(&mut self, t: T, mut f: F)
    |                                ^

error[E0277]: the size for values of type `Self` cannot be known at compilation time
   --> /home/huitseeker/.cargo/git/checkouts/kompact-8e3803f644d2dc0c/b3d31ba/core/src/utils.rs:647:36
    |
647 |         fn for_each_try_with<T, F>(mut self, t: T, mut f: F) -> Result<(), SerError>
    |                                    ^^^^^^^^ doesn't have a size known at compile-time
    |
    = help: unsized fn params are gated as an unstable feature
help: consider further restricting `Self`
    |
650 |             F: FnMut(Self::Item, T), Self: Sized
    |                                    ^^^^^^^^^^^^^
help: function arguments must have a statically known size, borrowed types always have a known size
    |
647 |         fn for_each_try_with<T, F>(&mut self, t: T, mut f: F) -> Result<(), SerError>
    |                                    ^

error: aborting due to 2 previous errors
```

## Workaround

One simple way to upgrade is to update the activation of the `unsized_locals` feature to target `unsized_fn_params`. Except this is not backwards-compatible (feature gate errors are fatal and this triggers E0635) and will break every nightly below 2020-10-29.

That includes most nightlies in interactive (human-facing) use since rustfmt has not compiled on nightly since 2020-10-25 or so. It's advised not to merge until the "main" components are available in a post 10-29 nightly here :
https://rust-lang.github.io/rustup-components-history/

--- a/core/src/lib.rs
+++ b/core/src/lib.rs
@@ -71,7 +71,7 @@
 #![cfg_attr(nightly, feature(never_type))]
 #![cfg_attr(nightly, feature(option_expect_none))]
 #![cfg_attr(nightly, feature(async_closure))]
-#![cfg_attr(nightly, feature(unsized_locals))]
+#![cfg_attr(nightly, feature(unsized_fn_params))] // requires nightly > 2020-10-29
 
 #[cfg(feature = "thread_pinning")]
 pub use core_affinity::{get_core_ids, CoreId};

